cel/canonical_eventlog_test.go

```diff
@@ -11,51 +11,66 @@ func TestCELEncodingDecoding(t *testing.T) {
 	hashAlgoList := []crypto.Hash{crypto.SHA256, crypto.SHA1, crypto.SHA512}
 	cel := &CEL{}
 
-	someEvent := make([]byte, 10)
-	cosEvent := CosTlv{someEvent}
-	err := cel.AppendEvent(nil, 13, hashAlgoList, cosEvent)
-	if err != nil {
-		t.Fatal(err.Error())
+	testEvents := []struct {
+		cosNestedEventType uint8
+		pcr                int
+		eventPayload       []byte
+	}{
+		{ImageRefType, 13, []byte("docker.io/bazel/experimental/test:latest")},
+		{ImageDigestType, 14, []byte("sha256:781d8dfdd92118436bd914442c8339e653b83f6bf3c1a7a98efcfb7c4fed7483")},
+		{RestartPolicyType, 15, []byte("Never")},
 	}
 
-	cosEvent2 := CosTlv{someEvent}
-	err = cel.AppendEvent(nil, 14, hashAlgoList, cosEvent2)
-	if err != nil {
-		t.Fatal(err.Error())
+	for _, testEvent := range testEvents {
+		cos := CosTlv{testEvent.cosNestedEventType, testEvent.eventPayload}
+		err := cel.AppendEvent(nil, testEvent.pcr, hashAlgoList, cos)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
 	}
 
 	var buf bytes.Buffer
-	err = cel.EncodeCEL(&buf)
+	err := cel.EncodeCEL(&buf)
 	if err != nil {
 		t.Fatalf(err.Error())
 	}
 	decodedcel, err := DecodeToCEL(&buf)
 	if err != nil {
 		t.Fatalf(err.Error())
 	}
-	if len(decodedcel.Records) != 2 {
-		t.Errorf("should have two records")
-	}
-	if decodedcel.Records[0].RecNum != 0 {
-		t.Errorf("recnum mismatch")
-	}
-	if decodedcel.Records[1].RecNum != 1 {
-		t.Errorf("recnum mismatch")
-	}
-	if decodedcel.Records[0].PCR != 13 {
-		t.Errorf("pcr value mismatch")
-	}
-	if decodedcel.Records[1].PCR != 14 {
-		t.Errorf("pcr value mismatch")
+	if len(decodedcel.Records) != 3 {
+		t.Errorf("should have three records")
 	}
 
-	digestsMap := decodedcel.Records[0].Digests
-	if len(digestsMap[crypto.SHA256]) != 32 {
-		t.Errorf("SHA256 digest length doesn't match")
-	}
-	if len(digestsMap[crypto.SHA1]) != 20 {
-		t.Errorf("SHA1 digest length doesn't match")
+	for i, testEvent := range testEvents {
+		if decodedcel.Records[i].RecNum != uint64(i) {
+			t.Errorf("recnum mismatch for record %d", i)
+		}
+		if decodedcel.Records[i].PCR != uint8(testEvent.pcr) {
+			t.Errorf("pcr value mismatch for record %d", i)
+		}
+		if len(decodedcel.Records[i].Digests[crypto.SHA512]) != 64 {
+			t.Errorf("SHA256 digest length doesn't match for record %d", i)
+		}
+		if len(decodedcel.Records[i].Digests[crypto.SHA256]) != 32 {
+			t.Errorf("SHA256 digest length doesn't match for record %d", i)
+		}
+		if len(decodedcel.Records[i].Digests[crypto.SHA1]) != 20 {
+			t.Errorf("SHA1 digest length doesn't match for record %d", i)
+		}
+
+		extractedCos, err := NewCosTlvFromTlv(decodedcel.Records[i].Content)
```

Comment:
seems like we're testing both CEL encode decode with COS functionality in this test. let's break it up into another test in a `cos_tlv_test.go` file